sfge/src/main/java/com/salesforce/cli/Result.java

```diff
@@ -0,0 +1,62 @@
+package com.salesforce.cli;
+
+import com.salesforce.rules.Violation;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/** Represents the result of an SFGE execution. */
+public class Result {
+
+    /** Violations detected during an execution * */
+    private final List<Violation> violations;
+
+    /** Exceptions that interrupted the execution * */
+    private final List<Throwable> errorsThrown;
+
+    /** Indicates if the execution completed successfully * */
+    private boolean completedSuccessfully;
+
+    public Result() {
```

Comment:
Proposal: If the intended use for this object is that instances with results will be merged into instances without results, do we maybe want to have a static factory method called something like `getEmptyResults()`, and have the constructor be private?
Or would that be too weird?

Reply:
`merge()` turned out to be an afterthought.
The original intention was to make up for the capability to both return violations and throw exceptions. Since java allows only one of them, I had to resort to this bean class to hold both kinds of information. Making this a utility class could be weird, but I'd been alternating this side and that side to make a method like
```
public static Result merge(Result result1, Result result2)
```

I finally decided to make this an instance method since this would probably happen when SFGE runs into low heap space, and creating a third Result object to merge might make us lose any salvaged result.